src/index.ts

```diff
@@ -0,0 +1,40 @@
+//1
+interface Interface1 {
+	[key: string]: string | number;
+}
+
+interface Interface2 {
+	[key: number]: (() => void);
+}
+
+interface Interface3 {
+	[key: number]: string; // не зовсiм зрозумiв "...а значення - певного типу." Якого саме?
```

Comment:
Whatever value types in the array you want to achieve :-)

The array is a data structure. In common usage, an array is a specific type of object with numerical keys and values. We are attempting to replicate an array with various item types.

Reply:
ok, thank's